Userland/Libraries/LibMainStatic/Main.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Format.h>
+#include <AK/StringView.h>
+#include <AK/Vector.h>
+#include <LibC/sys/internals.h>
+#include <LibC/unistd.h>
+#include <LibMain/Main.h>
+#include <string.h>
+#include <time.h>
+
+namespace Main {
+
+static int s_return_code_for_errors = 1;
+
+int return_code_for_errors()
+{
+    return s_return_code_for_errors;
+}
+
+void set_return_code_for_errors(int code)
+{
+    s_return_code_for_errors = code;
+}
+
+}
+
+char* __static_environ[] = { nullptr }; // We don't get the environment without some libc workarounds..
+
+static void init_libc()
+{
+    environ = __static_environ;
+    __environ_is_malloced = false;
+    __stdio_is_initialized = false;
+    // Initialise the copy of libc included statically in Loader.so,
+    // initialisation of the dynamic libc.so is done by the DynamicLinker
+    __libc_init();
```

Comment:
`__libc_init()`should happen in crt0.cpp but it doesn't. 

I don't think any of this init code or the static environ should be required if we initialize the runtime properly there.

Reply:
It happens in the Dynamic loader:
https://github.com/SerenityOS/serenity/blob/42f36d4865d003bab2b19e4f378b1f30ec92cd47/Userland/DynamicLoader/main.cpp#L15-L23

Reply:
Yeah, this should be moved into the non-shared CRT file instead of relying on LibMain to do very essential initialization (although, the environment-related code should already be there, but it's probably not working due to missing LibC init).

Alternatively, we could also commonize the initialization that is done in the DynamicLoader and the initialization that should be done for non-shared binaries into a single CRT file that is shared for both types, but that should probably be kept back until the next round of toolchain rebuilds.